asn1sccserviceproviderinterface.h

```diff
@@ -0,0 +1,52 @@
+/****************************************************************************
+**
+** Copyright (C) 2017 N7 Mobile sp. z o. o.
+** Contact: http://n7mobile.pl/Space
+**
+** This file is part of ASN.1/ACN Plugin for QtCreator.
+**
+** Plugin was developed under a programme and funded by
+** European Space Agency.
+**
+** This Plugin is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** This Plugin is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#pragma once
+
+#include <QObject>
+
+#include <QNetworkReply>
+#include <QHash>
+
+#include "documentsourceinfo.h"
+
+namespace Asn1Acn {
+namespace Internal {
+
+class Asn1SccServiceProviderInterface {
```

Comment:
one of techniques to bypass need for (Interface) is to try to find "what is really a base class 'thing'" and "what's really different in implementation". So here I'd go with something like `ParsingServiceProvider` for iface and `Asn1SccServiceProvider` for child class - iface is just for parsing and usage of asn1scc is the special thing that separates implementation class from iface class.

I'd also consider not including "start" "stop" etc. in iface - then it could be named simple `ParsingService` with single `requestAst`. "start" etc. would be added by `Asn1SccServiceProvider` (or `Asn1SccService`) and used only by single class (probably plugin which will create it).